File: assets/languages/python/event-port/src/delivery_starter/application/ports/events.py
from dataclasses import dataclass
from typing import Literal, Protocol


@dataclass(frozen=True)
class DomainEvent:
    type: str
    schema_version: Literal[1]
    stream_id: str
    payload: dict[str, object]


@dataclass(frozen=True)
class AppendResult:
    outcome: Literal["appended", "version-conflict"]
    version: int


class EventStore(Protocol):
    def read(self, stream_id: str) -> tuple[DomainEvent, ...]: ...

    def append(
        self,
        stream_id: str,
        expected_version: int,
        events: tuple[DomainEvent, ...],
    ) -> AppendResult: ...

File: assets/languages/go/event-port/application/ports/events/events.go
package events

type DomainEvent struct {
	Type          string
	SchemaVersion int
	StreamID      string
	Payload       map[string]any
}

type AppendResult struct {
	Outcome string
	Version int
}

type Store interface {
	Read(streamID string) []DomainEvent
	Append(streamID string, expectedVersion int, events []DomainEvent) AppendResult
}
